### ISSUE TITLE
 Change error vars (e.g. BAD_SUBJECT_ERR) to strings

### ISSUE THREAD

**Issue by shaneiseminger** [nats-io/node-nats#89]
Working out an issue implementing the node-nats-streaming library and I was initially confused by the stack trace until I realized that it's because the `Error` objects are being instantiated as soon as the library was required, causing the stack trace to show the trace to the point the objects are instantiated, not the point that they're thrown.

Obvious suggestion is to set the error vars to strings and instantiate `Error` when throwing, e.g. `throw(new Error(BAD_SUBJECT_ERR))`. This change also seems slightly more efficient, as right now you have error objects always hanging around in memory whether they're needed or not.

Before this change, the stack trace looks like this:

````
<path>/node_modules/node-nats-streaming/node_modules/nats/lib/nats.js:938
      throw(BAD_SUBJECT_ERR);
      ^
Error: Subject must be supplied
    at Object.<anonymous> (<path>/node_modules/node-nats-streaming/node_modules/nats/lib/nats.js:70:23)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (<path>/node_modules/node-nats-streaming/node_modules/nats/index.js:1:80)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (<path>/node_modules/node-nats-streaming/lib/stan.js:15:12)
    at Module._compile (module.js:409:26)
````

After the suggested change:

````
Error: Subject must be supplied
    at Client.publish (<path>/node_modules/node-nats-streaming/node_modules/nats/lib/nats.js:938:13)
    at Client.request (<path>/node_modules/node-nats-streaming/node_modules/nats/lib/nats.js:1116:8)
    at Stan.subscribe (<path>/node_modules/node-nats-streaming/lib/stan.js:490:11)
[ stack trace from my code truncated ]
````

As you can see, the second stack trace is informative, the first is not.

Happy to submit a pull request for this unless you want to handle it a different way.

**Comment by aricart** [comment]
Hi there thanks for the report. This issue as you have found above, affects both node-nats and node-nats-streaming. The issue should be fairly easy to fix, since we can easily identify all the throw points. Let me do a fix.

Thanks for the report.

**Comment by shaneiseminger** [comment]
Thanks!

**Comment by aricart** [comment]
Fix for node-nats-streaming is ed6bdc7ea965eeeaa3b693451c9ca8043c0d2d4b
